sklearn/utils/__init__.py

```diff
@@ -319,6 +319,15 @@ def _check_key_type(key, superclass):
     return False
 
 
+def _is_negative_indexing(key):
+    def is_neg(x): isinstance(x, int) and x is not None and x < 0
```

Comment:
this returns None?

Are we ever going to need this apart for the current deprecation? I'd move that to the CT file and mark it to be removed in 0.24

Also I think we use numbers.Integral?

no need to check for None if `isinstance` passes.